service/test/controllers/search/SearchControllerTest.java

```diff
@@ -39,39 +41,60 @@
 import util.RequestInterceptor;
 
 /** Created by arvind on 6/12/17. */
-@FixMethodOrder(MethodSorters.NAME_ASCENDING)
 @RunWith(PowerMockRunner.class)
-@PrepareForTest(RequestInterceptor.class)
+@PrepareForTest({OnRequestHandler.class})
+@SuppressStaticInitializationFor({"util.AuthenticationHelper", "util.Global"})
 @PowerMockIgnore("javax.management.*")
-@Ignore
 public class SearchControllerTest {
 
-  private static play.Application app;
-  private static Map<String, List<String>> headerMap;
-  private static ActorSystem system;
-  private static final Props props = Props.create(DummyActor.class);
+  public static Application application;
+  public static ActorSystem system;
+  public static final Props props = Props.create(DummyActor.class);
 
-  @BeforeClass
-  public static void startApp() {
-    app = Helpers.fakeApplication();
-    Helpers.start(app);
-    headerMap = new HashMap<String, List<String>>();
-    headerMap.put(HeaderParam.X_Consumer_ID.getName(), Arrays.asList("Service test consumer"));
-    headerMap.put(HeaderParam.X_Device_ID.getName(), Arrays.asList("Some Device Id"));
-    headerMap.put(
-        HeaderParam.X_Authenticated_Userid.getName(), Arrays.asList("Authenticated user id"));
-    headerMap.put(JsonKey.MESSAGE_ID, Arrays.asList("Unique Message id"));
+  @Before
+  public void before() {
+    application =
+            new GuiceApplicationBuilder()
+                    .in(new File("path/to/app"))
+                    .in(Mode.TEST)
+                    .disable(StartModule.class)
+                    .build();
 
+    Helpers.start(application);
     system = ActorSystem.create("system");
     ActorRef subject = system.actorOf(props);
     BaseController.setActorRef(subject);
+    PowerMockito.mockStatic(OnRequestHandler.class);
+    Map<String, Object> inner = new HashMap<>();
+    Map<String, Object> aditionalInfo = new HashMap<String, Object>();
+    aditionalInfo.put(JsonKey.START_TIME, System.currentTimeMillis());
+    inner.put(JsonKey.ADDITIONAL_INFO, aditionalInfo);
+    Map outer = PowerMockito.mock(HashMap.class);
+    OnRequestHandler.requestInfo = outer;
+    PowerMockito.when(OnRequestHandler.requestInfo.get(Mockito.anyString())).thenReturn(inner);
+  }
+
+  @Test
+  public void testsyncWithKeyCloakOperation() {
+    Map<String, Object> requestMap = new HashMap<>();
+    Map<String, Object> innerMap = new HashMap<>();
+    innerMap.put(JsonKey.OBJECT_TYPE, JsonKey.ORGANISATION);
+    innerMap.put(JsonKey.OPERATION_FOR, "keycloak");
+    requestMap.put(JsonKey.REQUEST, innerMap);
```

Comment:
course service doesn't have flow to sync organisation or keycloak, if these are part of controller code, it should be cleaned. Test case should do syncing only for course batch or user courses